vm/src/builtins/set.rs

```diff
@@ -485,30 +485,60 @@ impl PySet {
 
     #[pymethod(name = "__ror__")]
     #[pymethod(magic)]
-    fn or(&self, other: SetIterable, vm: &VirtualMachine) -> PyResult<Self> {
-        self.union(other.iterable, vm)
+    fn or(&self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult<PyArithmeticValue<Self>> {
+        if other.isinstance(&vm.ctx.types.set_type)
+            || other.isinstance(&vm.ctx.types.frozenset_type)
+        {
+            let val = PosArgs::new(vec![ArgIterable::try_from_object(vm, other)?]);
+            Ok(PyArithmeticValue::Implemented(self.union(val, vm)?))
+        } else {
+            Ok(PyArithmeticValue::NotImplemented)
+        }
```

Comment:
`SetIterable` does most of this, it just doesn't return the right things. I think taking inspiration from what `PyInt` does (with `int_op`) would be the best thing here (a function that tries to create a `SetIterable` from the `PyObject` ref and then forwards the call to the appropriate set/frozenset method).

Reply:
thank you! I will refer to the information you provided and try to correct it! 🙏🏼

Reply:
I have a question. As you said, I referred to how int_op works in PyInt, which means that PySet should add functions such as set_op to PySet to implement similarly? Or does it mean adding another function to the SetItable structure?